x-pack/platform/plugins/shared/alerting/server/alert_deletion/lib/preview.ts

```diff
@@ -30,10 +30,7 @@ export const previewTask = async (
     }
   }
 
-  const ruleTypes =
-    categoryIds && categoryIds.length > 0
-      ? context.ruleTypeRegistry.getAllTypesForCategories(categoryIds)
-      : context.ruleTypeRegistry.getAllTypes();
+  const ruleTypes = context.ruleTypeRegistry.getAllTypesForCategories(categoryIds);
```

Comment:
not sure if this is the best approach @ymao1

Reply:
Yea that makes sense